src/prefect/client/client.py

```diff
@@ -284,7 +284,7 @@ def tenant_id(self) -> str:
 
     @tenant_id.setter
     def tenant_id(self, tenant_id: Union[str, uuid.UUID, None]) -> None:
-        if tenant_id is None:
+        if tenant_id in (None, ""):
```

Comment:
Just a style preference here
```suggestion
        if not tenant_id:
```